src/main/java/ch/wisv/areafiftylan/security/authentication/CurrentUserServiceImpl.java

```diff
@@ -95,6 +95,19 @@ public boolean canEditTeam(Object principal, Long teamId) {
     }
 
     @Override
+    public boolean canDeleteTeam(Object principal, Long teamId) {
```

Comment:
It feels a bit wrong to involve the state of a Team in this service, which should deal with User authorizations. I propose we move this logic to the Controller. It is a matter of authorization (which controllers handle) so I think some controller logic is permitted there. You can use the canEditTeam() PreAuthorize for that method instead.